doc/developer/setup/index.md

```diff
@@ -122,13 +122,56 @@ docker run hello-world
+However, if you are using cloud based storage you will still need to set these directories in the configuration as these will make up the virtual path to the file, but no local directories will be created.
+
+To setup IRIDA to use cloud based file storage, follow the instructions below for the storage type.
+
+### Setup using Azure Storage Blob
+
+In the configuration file (such as irida.conf) you will need to add these configuration values:
+
+* `irida.storage.type=azure`
+* `azure.container.name=CONTAINER_NAME` where the CONTAINER_NAME is a container previously setup on Azure
+* `azure.container.url=CONTAINER_ENDPOINT_URL`
+* `azure.sas.token=SAS_TOKEN` where the SAS_TOKEN has both read/write permissions
+
+See [Azure Storage Setup](https://learn.microsoft.com/en-us/azure/storage/blobs/) for instructions on how to setup Blob storage.
+
+Microsoft has also made available a storage emulator,`Azurite`, which can be used to develop and test Azure storage functionality on a local machine instead of requiring the use of an Azure Storage account. See [Microsoft Azurite](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite).
+
+For you to be able to use `Azurite` you will need to get the `SAS` token for the blob container. This can be retrieved by setting up [Azure Storage Explorer](https://learn.microsoft.com/en-us/azure/vs-azure-tools-storage-manage-with-storage-explorer) and adding a new resource (`Local Storage Emulator`). Once that is set-up, click on the `Blob Containers menu` item in the explorer and then right click on the container (created by default) `test`, and click Get Shared Access Signature. From the popup window, you can select the date range of validity of the token and permissions (Read, Write, Delete, List, Add, and Create) for the container, and then click create.
+
+Once you have Azurite setup and container created, you can update these configuration values (in irida.conf etc)
+* `irida.storage.type=azure`
+* `azure.container.name=test`
+* `azure.container.url=http://127.0.0.1:10000/devstoreaccount1/test?SAS_TOKEN_RETRIEVED_ABOVE`
+* `azure.sas.token=SAS_TOKEN_RETRIEVED_ABOVE` where the SAS_TOKEN has both read/write permissions
+
+If using `Azurite` make sure you have it running before starting up IRIDA.
+
+### Setup using Amazon AWS S3 Bucket Storage
+
+In the configuration file (such as irida.conf) you will need to add these configuration values:
+
+* `irida.storage.type=aws`
+* `aws.bucket.name=BUCKET_NAME` where the BUCKET_NAME is the S3 Bucket previously setup and has read/write permissions.
+* `aws.bucket.region=BUCKET_REGION`
+* `aws.access.key=ACCESS_KEY`
+* `aws.secret.key=SECRET_KEY`
+
+See [AWS S3 Bucket Storage Setup](https://docs.aws.amazon.com/AmazonS3/latest/userguide/GetStartedWithS3.html) for instructions on how to setup S3 storage.
```

Comment:
I did one quick Azure test using Azurite and it worked. :smile: 
Then I wanted to try testing AWS, but noticed you only give a link to the actual S3 documentation. I know you can sign up for a free trial, but I wasn't really happy that they asked for a credit card. Could we be use a AWS mock service, like Azurite? I found LocalStack, but then realised your settings don't take a url for AWS like they do for Azure. Is there anyway we could change that so developers can use a mock service of their choosing for development?

Reply:
I take that back. I see the version of the maven package we are using has a setEndpoint method. This will need to be tested out so either way I don't think we will get it into this release. Thanks again @ksierks

Reply:
I think you have to create an EndpointConfiguration. If not, I don't know how else to test AWS.

Reply:
Yeah I've been trying to get it to work. It'd be really helpful for testing it out. I'll keep working away at it and hopefully get it working! Thanks!